src/olympia/api/tests/test_permissions.py

```diff
@@ -215,105 +202,179 @@ def setUp(self):
+
+            # As above it doesn't work with the object though.
+            assert not self.permission.has_object_permission(
+                request, myview, obj)
+
+
+class TestAllowUnlistedReviewer(TestCase):
     # Note: be careful when testing, under the hood we're using a method that
     # relies on UserProfile.groups_list, which is cached on the UserProfile
     # instance.
     def setUp(self):
         self.permission = AllowReviewerUnlisted()
-        self.request = RequestFactory().get('/')
+        self.request = RequestFactory().post('/')
 
     def test_user_cannot_be_anonymous(self):
         self.request.user = AnonymousUser()
-        obj = Mock()
-        obj.is_listed = False
+        obj = Mock(spec=[])
+        obj.has_unlisted_versions = lambda: True
         assert not self.permission.has_permission(self.request, myview)
         assert not self.permission.has_object_permission(
             self.request, myview, obj)
 
     def test_authenticated_but_not_reviewer(self):
-        self.request.user = UserProfile.objects.get(pk=999)
-        obj = Mock()
-        obj.is_listed = False
+        self.request.user = user_factory()
+        obj = Mock(spec=[])
+        obj.has_unlisted_versions = lambda: True
         assert not self.permission.has_permission(self.request, myview)
         assert not self.permission.has_object_permission(
             self.request, myview, obj)
 
     def test_admin(self):
-        self.request.user = UserProfile.objects.get(email='admin@mozilla.com')
-        obj = Mock()
-        obj.is_listed = False
+        self.request.user = user_factory()
+        self.grant_permission(self.request.user, '*:*')
+        obj = Mock(spec=[])
+        obj.has_unlisted_versions = lambda: True
 
         assert self.permission.has_permission(self.request, myview)
         assert self.permission.has_object_permission(self.request, myview, obj)
 
+    def test_regular_reviewer(self):
+        self.request.user = user_factory()
+        self.grant_permission(self.request.user, 'Addons:Review')
+        obj = Mock(spec=[])
+        obj.has_unlisted_versions = lambda: True
+
+        assert not self.permission.has_permission(self.request, myview)
+        assert not self.permission.has_object_permission(
+            self.request, myview, obj)
+
     def test_unlisted_reviewer(self):
-        self.request.user = UserProfile.objects.get(
-            email='senioreditor@mozilla.com')
-        obj = Mock()
-        obj.is_listed = False
+        self.request.user = user_factory()
+        self.grant_permission(self.request.user, 'Addons:ReviewUnlisted')
+        obj = Mock(spec=[])
+        obj.has_unlisted_versions = lambda: True
 
         assert self.permission.has_permission(self.request, myview)
         assert self.permission.has_object_permission(self.request, myview, obj)
 
+    def test_object_with_no_unlisted_versions_and_listed_versions(self):
```

Comment:
nit: the method name is ambiguous on its own (yay English).  `test_object_with_listed_versions_but_no_unlisted_versions` ?